mlir/include/mlir/IR/BuiltinAttributes.h

```diff
@@ -1046,6 +1046,92 @@ class DistinctAttr
   static DistinctAttr create(Attribute referencedAttr);
 };
 
+//===----------------------------------------------------------------------===//
+// LoadedURIDenseResourceAttr
+//===----------------------------------------------------------------------===//
+
+namespace detail {
+struct LoadedURIDenseResourceAttrStorage;
+} // namespace detail
+
+/// An attribute that parses/prints as a identifier (file & symbol) while in
+/// memory representing the loaded attribute.
+
+class LoadedURIDenseResourceAttr
```

Comment:
Can this be mostly defined in ODS?

Reply:
Yes, I was struggling with it a bit. Its been a while ... hence figured get this for discussion and refactor in mechanical way.